### PR TITLE
file: refuse "getting" a directory

### LINKED ISSUE #6379: curl fails to resolve file URIs pointing to directories on Windows
First -- happy holidays! I hope the season is treating you well.

tl;dr: On *nix machines, it looks like `curl file:///path/to/directory` can be used to query the existence of a directory, while that same sort of invocation fails on Windows. I'm calling this a bug just because the behavior differs between platforms; please feel free to close with prejudice if you disagree!

#### macOS

```
$ curl -I file://$HOME
Content-Length: 5440
Accept-ranges: bytes
Last-Modified: Fri, 25 Dec 2020 07:07:09 GMT
```

#### Windows

```
C:\Users\kevinushey\bin>curl.exe -I file:///C:/Users/kevinushey/bin
curl: (37) Couldn't open file C:/Users/kevinushey/bin

C:\Users\kevinushey\bin>curl.exe -I file:///C:/Users/kevinushey/bin/curl.exe
Content-Length: 4395640
Accept-ranges: bytes
Last-Modified: Sat, 26 Dec 2020 05:35:49 GMT
```

### I expected the following

- A 0 status code (success) when attempting to resolve an existing directory;
- Headers describing the directory size + modified time if available.

### curl/libcurl version

```
C:\Users\kevinushey\bin>curl -V
curl 7.74.0 (x86_64-pc-win32) libcurl/7.74.0 OpenSSL/1.1.1i (Schannel) zlib/1.2.11 brotli/1.0.9 zstd/1.4.8 WinIDN libssh2/1.9.0 nghttp2/1.42.0
Release-Date: 2020-12-09
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile MultiSSL NTLM SPNEGO SSL SSPI TLS-SRP Unicode UnixSockets alt-svc brotli libz zstd
```

### operating system

Windows 10 Build 19042.

### PR DESCRIPTION
**Pull request by bagder**
Make non-windows platforms also return error if the given path is a
directory. curl can't transfer a directory and doing a "-I" on it would
return a "bogus" Content-Length on it.

Reported-by: Kevin Ushey
Fixes #6379

### CODE REVIEW

**Comment by bagder** [comment]
I'm ditching this. We should probably rather go the reverse and support dirs on Windows too...
